app/src/main/java/uk/ac/ebi/atlas/experimentpage/cellplot/CellPlotService.java

```diff
@@ -103,27 +103,30 @@ public ImmutableSet<TSnePoint> expressionPlot(String experimentAccession,
     }
 
     public ImmutableMap<String, JsonObject> fetchDefaultPlotMethodWithParameterisation(String experimentAccession) {
-        var umap = "umap";
-        var tsne = "tsne";
+
         var defaultcellPlots = cellPlotDao.fetchDefaultPlotMethodWithParameterisation(experimentAccession);
 
         ImmutableMap.Builder<String, JsonObject> defaultPlotTypeAndOptions = new ImmutableMap.Builder<>();
-        if (!defaultcellPlots.isEmpty()) {
-            defaultPlotTypeAndOptions.put(umap, getMiddleElement(defaultcellPlots.get(umap)));
-            defaultPlotTypeAndOptions.put(tsne, getMiddleElement(defaultcellPlots.get(tsne)));
-        }
+
+        defaultcellPlots.forEach((method, options) -> defaultPlotTypeAndOptions.put(method, getMiddleElement(defaultcellPlots.get(method))));
+
         return defaultPlotTypeAndOptions.build();
     }
 
-    private static JsonObject getMiddleElement(List plotOptions) {
-
-        if (plotOptions.size() % 2 == 0) { // even number
-            Object umapEvenItem = plotOptions.get((plotOptions.size() / 2 - 1));
-            return (JsonObject) umapEvenItem;
-        } else { //odd number
-            Object umapOddItem = plotOptions.get((plotOptions.size() / 2));
-            return (JsonObject) umapOddItem;
+    private JsonObject getMiddleElement(List plotOptions) {
+
+        if (plotOptions.isEmpty() || plotOptions == null) {
```

Comment:
if `plotOptions` is `null` this code is going to fail as you can't call a method on a null object, in this case `isEmpty()`. Please change the condition as I suggest above. You can just apply my suggestion.

```suggestion
        if (plotOptions == null || plotOptions.isEmpty()) {
```